browser/search_engines/search_engine_provider_service_factory.cc

```diff
@@ -25,14 +25,13 @@ KeyedService* InitializeSearchEngineProviderServiceIfNeeded(Profile* profile) {
   // This controller monitor's normal profile's service and apply it's change to
   // otr profile to use same provider.
   // Private profile's setting is shared with normal profile's setting.
-  if (profile->GetProfileType() == Profile::INCOGNITO_PROFILE) {
+  if (profile->IsIncognitoProfile()) {
     return new PrivateWindowSearchEngineProviderService(profile);
   }
 
   // Regardless of qwant region, tor profile needs controller to store
   // previously set search engine provider.
-  if (profile->IsTorProfile() &&
-      profile->GetProfileType() == Profile::GUEST_PROFILE) {
+  if (profile->IsTorProfile() && IsGuestProfile(profile)) {
```

Comment:
should the `IsGuestProfile()` call be moved into `IsTorProfile()`? Currently, Tor is using the guest profile... but I think @yrliou is working on a new Tor profile based on the Incognito profile. If what constitutes a Tor Profile changes, it would be easier to change in a "central" place

Reply:
I believe, currently `IsTorProfile` would return true for both the regular and off-the-record profiles of Tor profile. I am not sure what effect it would have to make it true only for off-the-record side. Would have to make sure that no place where it is called actually wants the regular side.

Reply:
no worries- since that would be a change to the behavior, let's not make it now
cc: @yrliou for visibility (in case we wanted to change in the future)